internal/util/telemetry/reporter.go

```diff
@@ -239,8 +239,17 @@ func makeRequest(s *state.State, m *connmetrics.ConnMetrics) *request {
 // with update available and latest version if any update is available.
 func (r *Reporter) report(ctx context.Context) {
 	s := r.P.Get()
+
 	if s.Telemetry != nil && !*s.Telemetry {
 		r.L.Debug("Telemetry is disabled, skipping reporting.")
+
+		// If telemetry is disabled, we can't say anything about update availability, but we keep latest version info.
+		if err := r.P.Update(func(s *state.State) {
+			s.UpdateAvailable = false
```

Comment:
We should do when telemetry is disabled, when one day passes and send a report or not